testsuite/features/step_definitions/common_steps.rb

```diff
@@ -510,10 +510,10 @@
   get_target('server').run_until_ok("ls /var/cache/rhn/repodata/#{channel}/*updateinfo.xml.gz")
 end
 
-When(/^I push package "([^"]*)" into "([^"]*)" channel$/) do |arg1, arg2|
-  srvurl = "https://#{get_target('server').full_hostname}/APP"
-  command = "rhnpush --server=#{srvurl} -u admin -p admin --nosig -c #{arg2} #{arg1} "
+When(/^I push package "([^"]*)" into "([^"]*)" channel$/) do |package, channel|
+  command = "rhnpush -u admin -p admin --nosig -c #{channel} #{package}"
   get_target('server').run(command, timeout: 500)
+  # TODO: instead of next line, wait for package to appear inside /var/spacewalk/packages
```

Comment:
What does it mean, refering to `ls`?

Reply:
Yes, the "ls" command on next line does basically nothing.

I did not want to remove it though, in case it had the implicit effect to wait until the operation is done.

Reply:
Ok, instead of listing all the directory it is probably more suitable to search for specific entry in that directory, something like `find /var/spacewalk/packages -name #{package}`.
Feel free to close this comment anytime you want.

Reply:
That's exactly what I was trying to suggest in my TODO comment: wait for the package to appear. I did not specify "with a find" in case people have some other method in mind, but that was my idea.

Closing this conversation.